todo_list/todo_list_app/views.py

```diff
@@ -0,0 +1,111 @@
+from django.shortcuts import render, redirect
+from .models import Category, Task
+from django.db.models import Q
+from django.contrib import messages
+from django.contrib.auth import authenticate, login, logout
+from django import forms
+
+
+def login_user(request):
+    return render(request, 'auth/login.html', {})
+
+
+def index(request):
+    return render(request, 'todo_list_app/index.html')
+
+
+def all_categories(request):
+    query = request.GET.get('q')
+    category_list = Category.objects.all()
```

Comment:
Are we getting all categories here or is it not loading any object in memory until accessed?

Reply:
We should list out only the current user's categories.